src/cml.js

```
@@ -1,49 +1,58 @@
 const globby = require('globby');
+const git = require('simple-git/promise')('./');
 
 const Gitlab = require('./drivers/gitlab');
 const Github = require('./drivers/github');
 const BitBucketCloud = require('./drivers/bitbucket_cloud');
 const { upload, exec, watermark_uri } = require('./utils');
 
+const {
+  GITHUB_REPOSITORY,
+  CI_PROJECT_URL,
+  BITBUCKET_REPO_UUID,
+  CI
+} = process.env;
+
+const GIT_USER_NAME = 'iterative-olivaw';
+const GIT_USER_EMAIL = 'olivaw@iterative.ai';
+const GIT_REMOTE = 'origin';
+const GITHUB = 'github';
+const GITLAB = 'gitlab';
+const BB = 'bitbucket';
+
 const uri_no_trailing_slash = (uri) => {
   return uri.endsWith('/') ? uri.substr(0, uri.length - 1) : uri;
 };
 
-const repo_from_origin = () => {
-  const origin = execSync('git config --get remote.origin.url').toString(
+const git_remote_url = (opts = {}) => {
+  const { remote = GIT_REMOTE } = opts;
+  const url = execSync(`git config --get remote.${remote}.url`).toString(
     'utf8'
   );
-  const uri = git_url_parse(origin).toString('https').replace('.git', '');
-  return strip_auth(uri);
+  return strip_auth(git_url_parse(url).toString('https').replace('.git', ''));
 };
 
 const infer_driver = (opts = {}) => {
   const { repo } = opts;
-  if (repo && repo.includes('github.com')) return 'github';
-  if (repo && repo.includes('gitlab.com')) return 'gitlab';
-  if (repo && repo.includes('bitbucket.com')) return 'bitbucket';
+  if (repo && repo.includes('github.com')) return GITHUB;
+  if (repo && repo.includes('gitlab.com')) return GITLAB;
+  if (repo && repo.includes('bitbucket.com')) return BB;
 
-  const {
-    GITHUB_REPOSITORY,
-    CI_PROJECT_URL,
-    BITBUCKET_REPO_UUID
-  } = process.env;
-  if (GITHUB_REPOSITORY) return 'github';
-  if (CI_PROJECT_URL) return 'gitlab';
-  if (BITBUCKET_REPO_UUID) return 'bitbucket';
+  if (GITHUB_REPOSITORY) return GITHUB;
+  if (CI_PROJECT_URL) return GITLAB;
+  if (BITBUCKET_REPO_UUID) return BB;
```

should we throw an error after this?

---

its actually throwing

```js
throw new Error(`driver ${driver} unknown!`);
```